### PR TITLE
two tests for the extensions option

### PR DESCRIPTION
**Pull request by mightyiam**

### CODE REVIEW

**Comment by mightyiam** [comment]
Again, I thought that one of these was gonna fail, but nope. So here we go.

**Comment by mightyiam** [comment]
Never mind about failing. I've found my issue.

File: test/extensions.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	});
+	it("should resolve according to order of provided extensions (dir index)", function(done) {
+		resolver.resolve({}, fixture, "./dir", function(err, result) {
+			result.should.equal(path.resolve(fixture, "dir", "index.ts"));
+			done();
+		});
+	});
+	it("should resolve according to main field in module root", function(done) {
+		resolver.resolve({}, fixture, ".", function(err, result) {
+			result.should.equal(path.resolve(fixture, "index.js"));
```
Comment (mightyiam) [inline test/extensions.js]: So, this is the current behavior. the `main` field wins over extensions. I don't know if that's correct. It hit me hard until I noticed it.

Comment (sokra) [inline test/extensions.js]: If the main field already has an extensions, others are not used. `extensions` are only appended to the request and not exchanged.

i. e. `./abc` -> `abc`, `abc.js`, `abc.ts`.
i. e. `./abc.js` -> `abc.js`, `abc.js.js`, `abc.js.ts`

**Comment by sokra** [comment]
Thanks

**Pull request merged**